/*
*  This file is part of aasdk library project.
*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
*
*  aasdk is free software: you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 3 of the License, or
*  (at your option) any later version.

*  aasdk is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  You should have received a copy of the GNU General Public License
*  along with aasdk. If not, see <http://www.gnu.org/licenses/>.
*/

syntax="proto2";

import "VideoResolutionEnum.proto";
import "VideoFPSEnum.proto";
import "MediaCodecTypeEnum.proto";

package oaa.proto.data;

message VideoConfig
{
    required enums.VideoResolution.Enum video_resolution = 1;
    required enums.VideoFPS.Enum video_fps = 2;
    required uint32 margin_width = 3;
    required uint32 margin_height = 4;
    required uint32 dpi = 5;
    optional uint32 additional_depth = 6;
    // Fields 7-9 exist in APK but purpose unknown (7=500, 8=10000 in captures)
    optional uint32 field7 = 7;
    optional uint32 field8 = 8;
    optional uint32 field9 = 9;
    // Codec type — discovered via APK decompilation (wcz.java field 10, vyn.java enum)
    optional enums.MediaCodecType.Enum codec = 10;

    // Discovered via APK proto decoder (wcz.java → wcm sub-message)
    // Purpose unknown — possibly additional video config params
    optional bytes additional_config = 11;
}
